sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java

```diff
@@ -185,26 +188,27 @@ public CosmosItemResponse upsertItem(Object item, CosmosItemRequestOptions optio
      * @return the cosmos sync item response
      * @throws CosmosClientException the cosmos client exception
      */
-    CosmosItemResponse mapItemResponseAndBlock(Mono<CosmosAsyncItemResponse> itemMono)
-            throws CosmosClientException {
+    CosmosItemResponse mapItemResponseAndBlock(Mono<CosmosAsyncItemResponse> itemMono) throws CosmosClientException {
         try {
             return itemMono
-                           .map(this::convertResponse)
-                           .block();
+                       .map(this::convertResponse)
+                       .block();
         } catch (Exception ex) {
             final Throwable throwable = Exceptions.unwrap(ex);
             if (throwable instanceof CosmosClientException) {
                 throw (CosmosClientException) throwable;
             } else {
-                throw ex;
+                throw Exceptions.propagate(ex);
```

Comment:
If the exception is not an instance of RuntimeException, i.e. a checked exception, this statement `Exceptions.propagate(ex)` will throw ReactiveException wrapped around inner exception. 

We should make sure that we want that behavior, or we need to unwrap the exception wherever we depend on it. 

```
public static RuntimeException propagate(Throwable t) {
		throwIfFatal(t);
		if (t instanceof RuntimeException) {
			return (RuntimeException) t;
		}
		return new ReactiveException(t);
	}
```

Reply:
good point Kushagra. Thanks for catching this. Bhaskar can this be rolled back to what it used be?
ie. `throw ex`? here and in other places.

Reply:
Missed rolling this back. I do a quick look and will revert all changes around these

Reply:
Done